test/integration/test_driver.go.tmpl

```diff
@@ -0,0 +1,42 @@
+
+
+// FIXME: No parameter needed for this function.
+func createOpenAPIBuilderConfig(getDefinitions common.GetOpenAPIDefinitions) *common.Config {
+	return &common.Config{
+		ProtocolList:   []string{"https"},
+		IgnorePrefixes: []string{"/swaggerapi"},
+		Info: &spec.Info{
+			InfoProps: spec.InfoProps{
+				Title:   "Integration Test",
+				Version: "1.0",
+			},
+		},
+		GetDefinitions: getDefinitions,
+	}
+}
+
+func main() {
+
+	// Create the OpenAPI builder config for creating swagger.
+	// FIXME: Currently the builder will not generate the definitions
+	// if there are no routes with types as parameters.
+	// NOTE: GetOpenAPIDefinitions has already been created in generated code.
+	config := createOpenAPIBuilderConfig(GetOpenAPIDefinitions)
+	empty := []*restful.WebService{}
+	swagger, serr := builder.BuildOpenAPISpec(empty, config)
+	if serr != nil {
+		panic(fmt.Sprintf("ERROR: %s", serr.Error()))
+	}
+
+	// Marshal the swagger spec into JSON, then write it out.
+	specBytes, err := json.MarshalIndent(swagger, " ", " ")
+	if err != nil {
+		panic(fmt.Sprintf("ERROR: %s", err.Error()))
+	}
+	err = ioutil.WriteFile("./def.json", specBytes, 0644)
+	if err != nil {
+		panic(fmt.Sprintf("ERROR: %s", err.Error()))
+	}
+
+	// NEXT: compare the generated swagger to golden swagger.
```

Comment:
Disagree. Comparing to the golfen file should be done in the test, not here. The goal of this compiled file is merely to generate the `swagger.json`. We need to keep it as small as possible, since it's harder to debug etc ...

Reply:
Agree w/ @apelisse.